staging/src/k8s.io/csi-translation-lib/plugins/vsphere_volume.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	v1 "k8s.io/api/core/v1"
+	storage "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// VSphereDriverName is the name of the CSI driver for vSphere Volume
+	VSphereDriverName = "csi.vsphere.vmware.com"
+	// VSphereInTreePluginName is the name of the in-tree plugin for vSphere Volume
+	VSphereInTreePluginName = "kubernetes.io/vsphere-volume"
+	// Name of parameter to supply SPBM Policy name for Volume Provisioning
+	paramStoragePolicyName = "storagepolicyname"
+	// Name of parameter to let controller know that it needs to return volumefilePath as Volume Attribute
+	// When CSI Driver Provision Volume using in-tree Storage Class and PVC
+	paramRequestVolumeFilePath   = "requestvolumefilepath"
+	paramVolumeFilePathAttribute = "volumefilepath"
```

Comment:
Can we call this param as `initialvolumefilepath` or something similar to call out that this path will not be updated if the CNS volume is storage vmotioned to another datastore?

Reply:
Yes changing it to

```
// AttributeInitialVolumeFilepath represents the path of volume where volume is created
AttributeInitialVolumeFilepath = "initialvolumefilepath"
```